Expose darwinssl as an option when building curl

sgr-ms opened the pull request:
This exposes the ability to built curl while using Apple's SSL/TLS implementation.

kiwixz committed:
update to cmake 3.14 (allow VS2019RC)

kiwixz committed:
fix vs2019 arch specification

ras0219-msft committed:
[vcpkg] Update to CMake 3.14.0

ras0219-msft committed:
Merge remote-tracking branch 'origin/master' into HEAD

ras0219-msft committed:
[vcpkg] Revert change to freebsd cmake version

ras0219-msft committed:
[vcpkg] Fixup previous commit

ras0219-msft committed:
[vcpkg] Fixup previous commit

Commit:
[giflib] Changed library linkage to static.

colatkinson committed:
Update to 0.1.15, and update CMakeLists.txt

colatkinson committed:
Remove unnecessary patch to libnice

sgr-ms committed:
expose darwinssl as an option

sgr-ms committed:
expose darwinssl as an option

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=5824) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=5824) before we can accept your contribution.<br/><br/>:x: sgr-ms [sign now](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=5824)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/vcpkg?pullRequest=5824) it.</sub>

ras0219-msft committed:
[giflib] Modernize. Avoid use of WINDOWS_EXPORT_ALL_SYMBOLS.

past-due commented on ports/curl/portfile.cmake:
```diff
@@ -87,6 +98,7 @@ vcpkg_configure_cmake(
         -DCMAKE_USE_OPENSSL=${USE_OPENSSL}
         -DCMAKE_USE_WINSSL=${USE_WINSSL}
         -DCMAKE_USE_MBEDTLS=${USE_MBEDTLS}
+        -DCMAKE_USE_DARWINSSL=$USE_DARWINSSL}
```
Looks like a small typo here? `$USE_DARWINSSL}` -> `${USE_DARWINSSL}`

Rastaban committed:
Merge pull request #5811 from colatkinson/libnice-0.1.15

Update libnice to 0.1.15

Rastaban committed:
Merge pull request #5784 from qis/dev/qis/giflib

[giflib] Changed library linkage to static.

ewertons committed:
[azure-iot] vcpkg update for master/public-preview release (#5789)

* [azure-iot] vcpkg update for master/public-preview release

* [azure] Fix download hash strings

* [azure] Bump CONTROL versions

LilyWangL committed:
[many ports]Add long path warnings

sgr-ms committed:
Fix typo

sgr-ms committed:
Fix typo

eao197 committed:
[json-dto] updated to v.0.2.8 (#5836)

* sobjectizer updated to v.5.5.24.3

* json-dto updated to v.0.2.8

lazyhamster committed:
[libmspack] update to 0.10.1. (#5834)

Rastaban committed:
Merge pull request #5835 from wangli28/dev/Lily/LongPathWarning

[many ports]Add long path warnings

JackBoosY committed:
[mozjpeg]Fix installation files to the wrong directory (#5833)

* [jemalloc]Fix dll generated when generating static library

* Revert "[jemalloc]Fix dll generated when generating static library"

This reverts commit a833bf5593be181e065af097732c78cb810888fd.

* [mozjpeg]1.Fix installation files to the wrong directory.   2.Change how dynamic and static library built macros are used

BillyONeal committed:
Update Casablanca to v2.10.12. (#5830)

vicroms commented:
CI doesn't test non-default features. 
@Rastaban can you test installing `curl[darwinssl]` locally?

Commit:
[gdal] fix for building static library (#5839)

sgr-ms committed:
add an error in case darwinssl is used on a non darwin system

sgr-ms committed:
add an error in case darwinssl is used on a non darwin system

sgr-ms commented:
> 
> 
> CI doesn't test non-default features.
> @Rastaban can you test installing `curl[darwinssl]` locally?
> 
> @sgr-ms
> As this only works for Mac, I wonder if adding a validation to disable the feature for non-Mac should be added. Although, it would be wrong for a user to attempt installing the feature on any other platform.
> 
> `if("darwinssl" IN_LIST FEATURES AND VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin")`

Sure, I have added something similar to what winssl has.

Rastaban commented:
@vicroms sure, I will test it out later today after I finish setting up my MacBook.

past-due commented:
@Rastaban:

It would be nice if we could update the `curl[ssl]` feature to default to `darwinssl` on Apple platforms.

So:
```
Feature: ssl
Build-Depends: curl[openssl] (!windows), curl[winssl] (windows)
Description: Default SSL backend
```

could become something like:
```
Feature: ssl
Build-Depends: curl[openssl] (!windows&!macos), curl[winssl] (windows), curl[darwinssl] (macos)
Description: Default SSL backend
```

But is it currently possible to specify `macos`/`darwin` to filter only Apple platforms in the `CONTROL` file?

JackBoosY committed:
[jemalloc]Fix dll generated when generating static library (#5829)

* [jemalloc]Fix dll generated when generating static library

LarryIII committed:
[pcre2/caffe2] Fix space issue (#5827)

* [pcre2/caffe2] Fix space issue

Rastaban commented:
@past-due you can filter on Mac by using the name `osx`.  See the antlr4 CONTROL file as an example.

CJ-Johnson committed:
Removes string replacement on Abseil headers

Abseil now has all uses of min and max wrapped in parens to prevent macro expansion. This will continue to be the case going forward, so these replacements are no longer needed.

Also, the Github web UI is forcing an extra change that updates the return character to the proper Windows CR+LF

sgr-ms commented:
> this



> 
> 
> @past-due you can filter on Mac by using the name `osx`. See the antlr4 CONTROL file as an example.

What about if you also use this to build for ios (like me) ? Do I need to specify each OS (e.g. iphoneos and simulator) as well ?

Rastaban commented:
Currently that filtering is based on the name of the triplet file (although that may change sometime in the future).  So yes, unless your ios triplet file also has `osx` in its name it will also need to be listed.

Rastaban committed:
[curl] add darwinssl to curl[ssl] osx default

Rastaban committed:
[curl] add darwinssl to curl[ssl] osx default

donny-dont committed:
Allow additional environment variables to be set in a clean environment (#5650)

* Allow additional environment variables to be set in a clean environment

* Add documentation for VCPKG_KEEP_ENV_VARS

Rastaban committed:
[decimal-for-cpp] update to 1.16 (#5852)

cenit committed:
[libhungarian] add port (#5849)

* [libhungarian] add port

* [hungarian] fix cmake targets

myd7349 committed:
[liblzma] Fix for UWP build (#5607)

JackBoosY committed:
[giflib]fix compile error in x64-uwp (#5578)

* [giflib]fix compile error in x64-uwp

* [giflib]Modify the location of the CMakelists.txt changes in the patch to avoid patching failures.

* [giflib]Modify cmakelists.txt directly and delete the corresponding changes in the patch

* [giflib] Modernize. Avoid use of WINDOWS_EXPORT_ALL_SYMBOLS.

myd7349 committed:
[protobuf] Update to 3.7.1 (#5502)

* [protobuf] Update to 3.7.1

Commit:
[exiv2] Update exiv2 to 0.27 release, optionally support unicode on windows (#5454)

Rastaban committed:
[freetype] fix issue #5774 (#5853)

* load 0005-Fix-DLL-EXPORTS.patch

* Create 0005-Fix-DLL-EXPORTS.patch

* [freetype] cleanup whitespace

sgr-ms commented:
> 
> 
> Currently that filtering is based on the name of the triplet file (although that may change sometime in the future). So yes, unless your ios triplet file also has `osx` in its name it will also need to be listed.
> 
> For this PR I am okay with only taking `osx` into account in the curl[ssh] feature. If you build curl[ssh] for iphones you would get the existing behavior by default, if you want darwinssl then you could either build curl[darwinssl] explicitly or modify the local copy of the control file to take your iphone triplet into account. Official support for ios is on our backlog (although it keeps getting pushed back) when that happens I expect we would make changes to this and other similar ports.

@Rastaban Seems like this broke the CI verification ? Or is this unrelated?

zhihaoy committed:
[ampl-mp] Limit the scope of ICE workaround (#5870)

JackBoosY committed:
[signalrclient]Fix build error (#5868)

* [signalrclient]Fix build error: LNK2019 with openssl

NancyLi1013 committed:
[jsonnet] Fix white-space-path issue. (#5862)

JackBoosY committed:
[luajit]move luajit headers to the directory "include/luajit/" to avoid file conflicts. (#5863)

[luajit]move luajit headers to the directory "include/luajit/" to avoid file conflicts.

JackBoosY committed:
[abseil]Fix link error when using abseil library. (#5860)

* [jemalloc]Fix dll generated when generating static library

* Revert "[jemalloc]Fix dll generated when generating static library"

This reverts commit a833bf5593be181e065af097732c78cb810888fd.

* [abseil]Fix link error when using abseil library.

Rastaban committed:
Removes string replacement on Abseil headers (#5850)

Abseil now has all uses of min and max wrapped in parens to prevent macro expansion. This will continue to be the case going forward, so these replacements are no longer needed.

Also, the Github web UI is forcing an extra change that updates the return character to the proper Windows CR+LF

JackBoosY committed:
[pdal]fix missing vendor headers (#5846)

* [pdal]1.fix missing vendor headers.   2.modify the patch name and add the serial number.

Rastaban commented:
Rerunning CI validation, the failing test is from a flakey port.

Commit:
[glew] Do not build utilities (#5859)

* [glew] Do not build utilities

Commit:
[opencv] Make opencv build exportable (#5857)

* [opencv] Make opencv build exportable

* [opencv] bump version number

Commit:
[xerces-c] Add libraries required on osx to target using wrapper (#5858)

* [xerces-c] Add libraries required on osx to target using wrapper

* [xerces-c] Bump version number

Rastaban commented:
it looks like google-cloud-cpp:x64-osx is failing.  I won't be able to get to the logs for this until Monday.

JackBoosY committed:
[libharu]fix build failure. (#5628)

* [libharu]fix build failure.

* [libharu]Comment more code

myd7349 committed:
[brunocodutra-metal] Add new port (#5817)

* [brunocodutra-metal] Add new port

* [brunocodutra-metal] Rename

Commit:
Fix for shared build broken due to #5839 (#5882)

greg7mdp committed:
add parallel-hashmap package (#5893)

greg7mdp committed:
[parallel hashmap] increment release version (#5898)

* update version of package

Helco committed:
[libuv/uvw] Update to 1.27.0 (#5895)

* Update libuv to 1.27.0

* Update uvw to 1.15.0

markdegreef committed:
[fcl] fix linux build (#5582)

* [fcl] fix linux build

Set correct path of cmake config files

markdegreef committed:
[octomap] fix linux build (#5583)

* [octomap] fix linux build

Copy entire bin folder for linux, as it only contains the executables

* [octomap] remove debug binaries

Asozial committed:
gdal: fixed typo (#5904)

* gdal: fixed typo

* Update CONTROL file

SuperWig committed:
[effolkronium-random] Update to 1.3.0 (#5900)

MistGebaut committed:
sdl1: static build (#5903)

* sdl1: static build

* Update CONTROL file

kiwixz committed:
[libwebp] do not include all feature by default (#5886)

* [libwebp] do not include all feature by default

* [libwep] bump control version

ShelleyLake committed:
[osg] static build (#5913)

lrineau committed:
Update CGAL to version 4.14 (#5734)

* CGAL now depends also on Boost Multiprecision
* Update to release 4.14

dlech committed:
Fix vcpkg_find_acquire_program(NUGET) (#5871)

Set the ARCHIVE and NOEXTRACT variables so that downloading works. Also HASH is case-sensitive, so make it lower case.

vcpkg_find_acquire_program(NUGET) was failing with the following error message:

-- Downloading https://dist.nuget.org/win-x86-commandline/v4.8.1/nuget.exe...
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:159 (file):
  file DOWNLOAD error: cannot create directory
  'C:/Users/david/Documents/GitHub/vcpkg/downloads/temp/C:/Users/david/Documents/GitHub/vcpkg/downloads'
  - Specify file by full path name and verify that you have directory
  creation and file write privileges.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_find_acquire_program.cmake:255 (vcpkg_download_distfile)
  ports/gobject-introspection/portfile.cmake:20 (vcpkg_find_acquire_program)
  scripts/ports.cmake:71 (include)

vicroms committed:
Merge branch 'master' of https://github.com/Microsoft/vcpkg into pr/cmake_3_14

vicroms committed:
[sqlitecpp] Find sqlite3 includes from vcpkg

vicroms committed:
[libodb-sqlite] Find sqlite3 includes from vcpkg

vicroms committed:
[libmodbus] Remove duplicated header files

vicroms committed:
[darknet,openssl-windows,tbb] Rebuild on CI

dzenanz committed:
Fix double-conversion issue in ITK (#5115)

* Using newer version of ITK which fixes double-conversion problem

* Minor style update of ITK portfile

* Using newer version of ITK which fixes double-conversion problem

* [itk] Improve version string

* [itk][hdf5] Update ITK commit. Depend on eigen3. Bump hdf5 control version.

* [hdf5] Update CONTROL

* [netcdf-c] Disable parallel configure due to writes to source directory

* [hdf5] Disable parallel configure due to writes to source directory. Switch to tar.gz to prevent CRLF line endings.

ras0219-msft committed:
[vcpkg-fetch] Set git subpath to mingw32\bin\git.exe to fix execution within containers (#5878)

ras0219-msft committed:
[vtk] Fix flakiness by relying on vcpkg-provided dependencies
[libharu] Add patches required for vtk
[vcpkg-build-cmake] Add retry logic for mt.exe failures

ras0219-msft committed:
[hdf5][vtk] Add vcpkg-cmake-wrapper for HDF5 to fix vtk

Cheney-W committed:
[live555] Update the download path (#5919)

Rastaban committed:
[openssl-windows] partially back out moderinzation change

LilyWangL committed:
[openmesh]Update version to 8.0 (#5923)

Commit:
[Thrift] Updates 2019.03.29 (#5924)

angelmixu committed:
update muParser to v.2.2.6.1 (#5931)

Cheney-W committed:
[milerius-sfml-imgui] Fix find_package issue (#5930)

Commit:
[RocksDB] Updates 2019.03.29 (#5933)

LarryIII committed:
[date] Fix failures in UWP with latest commit (#5932)

grdowns committed:
Merge branch 'dev/roschuma/vtk' of https://github.com/ras0219-msft/vcpkg into dev/grdowns/5935

grdowns committed:
[hdf5] Set wrapper policy for older dependent libraries

grdowns committed:
Merge branch 'pr/cmake_3_14' of https://github.com/kiwixz/vcpkg into dev/grdowns/5702

djarek committed:
Fix installation failure when system Boost present (#5939)

Boost 1.70 introduces "native" cmake config scripts which are
accidentally used when installing a vcpkg package. This only occurs
when Boost 1.70 is installed in the system when the user uses
`vcpkg install`.

Signed-off-by: Damian Jarek <damian.jarek93@gmail.com>

Rastaban committed:
[qca] don't use port dir for temp (#5954)

* [qca] don't use port dir for temp

* [qca] bump control version

philjdf committed:
[arrow] Update Arrow to v0.13 (#5950)

* Update arrow to 0.13.0

* dynamic build working

* Both builds now work

* modifications to work on Linux

* Turn of jemalloc

zharkov-eu committed:
Update msgpack to 3.1.1 (#5945)

Neumann-A committed:
[fftw3] fix git patch (#5926)

* Update fftw3_arch_fix.patch

fix fftw3 arch fix

* bump version in fftw3 control

Commit:
[fix][ffmpeg] fixing linux build (#5917)

* limit windows dep libs to windows

* fixing configure patch

* fixing detect-openssl.patch

* adapting portfile to linux

* adding a linux-specific build sh

* add missing endif()

* bumping version

* adding extra check for windows environment

fradav committed:
highfive bump to 2.0 and macos fix (same as win32) (#5955)

* highfive bump to 2.0 and macos fix (same as win32)

* macos fix

vicroms committed:
[gdal] Apply conditional patching (#5940)

grdowns committed:
[folly] Link against appropriate lib for build type

Rastaban committed:
[glew][octomap][roaring] fix parallel config (#5958)

* [glew][octomap][roaring] fix parallel config

fix CMake Error: Could not open file for write in copy operation

* [anax] fix parallel config

cenit committed:
[giflib] enable ninja (#5963)

koprok committed:
Fix LLVM cmake modules (#5848)

* [llvm] Reduce memory usage during linking

* [llvm] Fix cmake modules path

past-due commented:
@Rastaban: PR #5968 should fix the breakage in `google-cloud-cpp:x64-osx` caused by this port.

grdowns committed:
Merge pull request #5935 from ras0219-msft/dev/roschuma/vtk

[vtk] Fix flaky builds by using system libraries

grdowns committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg into dev/grdowns/5702

Rastaban committed:
[freeimage] fixup dependencies (#5969)

coryan committed:
Update google-cloud-cpp to v0.8.0. (#5972)

kiwixz committed:
[ebml,matroska] update ebml to 1.3.7, fix installation on linux (#5981)

* [ebml] fix installation on linux

* [ebml] update to 1.3.7

* [ebml] fix config installation on linux

* [matroska] cleanup, fix linux compilation

rickertm committed:
Add port for solid3 (#5989)

pantor committed:
[inja] Update to v2.1.0 (#5985)

* Update CONTROL

* Update portfile.cmake

david-antiteum committed:
[opentracing] v1.5.1 (#5974)

* OpenTracing #5970

* No UWP. Static requires dynamic #5970
- Dynamic version can be build without the static one
- Static version can only be build when dynamic is also active
- UWP is not supported

* Linux support #5970

jwillemsen committed:
[ace] Upgrade to 6.5.5 (#5992)

* ports/ace/CONTROL:
    * ports/ace/portfile.cmake:

koprok committed:
[pixman] Fix build on x86-linux (#5998)

grdowns committed:
Merge pull request #5702 from kiwixz/pr/cmake_3_14

Update to cmake 3.14 (allow compilation with VS2019RC v142)

jwillemsen committed:
[ace] Add support for Visual Studio 2019 (#6001)

* ports/ace/CONTROL:
    * ports/ace/portfile.cmake:

xiao-ai committed:
update sqlite3 to 3.27.2 (#6002)

Commit:
[Tbb] Updates 2019.03.29 (#5928)

* [Tbb] Updates 2019.03.29

* Set the tbb  installed libs in TBBConfig.cmake

NancyLi1013 committed:
[g2o] Fix g2o:x64-windows failed with POST_BUILD_CHECKS_FAILED.#4098 (#5925)

* [g2o] fix g20:x64-windows failed with POST_BUILD_CHECKS_FAILED.#4098

* Add "VCPKG_USE_HEAD_VERSION" to judge if it is from HEAD.

ras0219-msft committed:
[angle] Fix regressions introduced in #4862: (#5922)

Reintroduces the unofficial:: targets.
PREFER_NINJA.
Small simplifications/modernizations.

ras0219-msft committed:
[vcpkg] Synchronize vcpkg-base with external repo (#5934)

past-due committed:
[curl] Fix `tool` feature (#5971)

- Use appropriate EXECUTABLE_SUFFIX to support non-Windows platforms
- Fixup `curl-target-release.cmake` to point to the new executable location (fixing the CURL::curl target)

past-due committed:
[google-cloud-cpp] Fix libcurl linkage (#5968)

* [google-cloud-cpp] Fix libcurl linkage

* [curl] Fix libcurl exported CMake target config (Windows)

edmBernard committed:
[soci] bump version (#5964)

cenit committed:
[darknet] add compatibility with OpenCV 4 (#6018)

past-due committed:
[curl] Bump CONTROL version (#6012)

LarryIII committed:
[sdl1]Fix sdl1 conflict with sdl2 (#6011)

xnorpx committed:
Use Opus CMake #4875 (#6007)

past-due commented:
@sgr-ms: Can you rebase this branch on the latest master? (With PR #5968 and PR #5971 merged, the `google-cloud-cpp:x64-osx` issues should be fixed.)

sgr-ms committed:
Merge branch 'dev/sgr-ms/curl-add-darwinssl' of https://github.com/sgr-ms/vcpkg into dev/sgr-ms/curl-add-darwinssl

sgr-ms commented:
@past-due Is there anything else I can do to merge this ?

vicroms commented:
I'm not sure what happened with this PR's git history, but it has become quite hard to review.

sgr-ms commented:
@vicroms 
I rebased it on the latest master at the time, maybe something went wrong there ?

past-due commented:
The pertinent commits should be:
095ce4e
1d8d9ad
6af7229
4ae524d

(Perhaps it was the merge commit that added all the latest master commits to the history of this PR?)

sgr-ms commented:
> 
> 
> The pertinent commits should be:
> [095ce4e](https://github.com/Microsoft/vcpkg/commit/095ce4e9392a83a2e4944635c9f0e2b43caa0fe9)
> [1d8d9ad](https://github.com/Microsoft/vcpkg/commit/1d8d9ad35a24e56c6ac8e4b8deb567019a64a5f8)
> [6af7229](https://github.com/Microsoft/vcpkg/commit/6af72299f3e0d9efc53596caddbd8ed04e6477f8)
> [4ae524d](https://github.com/Microsoft/vcpkg/commit/4ae524d7962b8253115c222d69c1f7bc3294f998)
> 
> (Perhaps it was the merge commit that added all the latest master commits to the history of this PR?)

Probably :\
Is there anything I can do to fix this ?

past-due commented:
I cherry-picked just the pertinent commits, and opened a new PR here: #6124 (to make it easier to review).
